lib/client.js

```diff
@@ -118,6 +119,21 @@ module.exports = class Client extends EventEmitter {
       const contentEncoding = headers['content-encoding']
       const statusCode = res.statusCode
 
+      // "If HTTP 307, follow the redirect"
+      if (statusCode === 307) {
+        const redirectURL = new URL(headers['location'])
+        return this._makeRequest(
+          Object.assign({}, opts, {
+            protocol: redirectURL.protocol,
```

Comment:
@vweevers sorry the update made the old conversation go away, but here's the line,
I was commenting from https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/307
it seems protocol can be change (but not method or body), can you check? thanks!